test/integration-tests/Uniswap.test.ts

```diff
@@ -1196,6 +1206,480 @@ describe('Uniswap V2 and V3 Tests:', () => {
+        // transfer the token to eve
+        await v3NFTPositionManager.transferFrom(bob.address, eve.address, tokenId)
+
+        // eve permits bob to spend the token
+        await v3NFTPositionManager.connect(eve).setApprovalForAll(bob.address, true)
+
+        // eve creates a signature for the router to spend the token
+        let { v, r, s } = await getPermitNFTSignature(eve, v3NFTPositionManager, router.address, tokenId, MAX_UINT)
+        planner.addCommand(CommandType.ERC721_PERMIT, [ADDRESS_THIS, tokenId, MAX_UINT, v, r, s])
+
+        await executeRouter(planner)
+        planner = new RoutePlanner()
+
+        let position = await v3NFTPositionManager.positions(tokenId)
+        let liquidity = position.liquidity
+
+        const params = { tokenId: tokenId, liquidity: liquidity, amount0Min: 0, amount1Min: 0, deadline: MAX_UINT }
+
+        const encodedDecreaseCall = encodeDecreaseLiquidity(params)
+
+        planner.addCommand(CommandType.V3_POSM_CALL, [encodedDecreaseCall])
+
+        await executeRouter(planner)
+      })
+    })
+
+    describe('collect liquidity', () => {
+      it('collect succeeds', async () => {
+        let bobToken0BalanceBefore = await usdcContract.balanceOf(bob.address)
+        let bobToken1BalanceBefore = await wethContract.balanceOf(bob.address)
+
+        // first we need to permit the router to spend the nft
+        let { v, r, s } = await getPermitNFTSignature(bob, v3NFTPositionManager, router.address, tokenId, MAX_UINT)
+        planner.addCommand(CommandType.ERC721_PERMIT, [ADDRESS_THIS, tokenId, MAX_UINT, v, r, s])
+
+        let position = await v3NFTPositionManager.positions(tokenId)
+        let liquidity = position.liquidity
+
+        const decreaseParams = {
+          tokenId: tokenId,
+          liquidity: liquidity,
+          amount0Min: 0,
+          amount1Min: 0,
+          deadline: MAX_UINT,
+        }
+
+        const encodedDecreaseCall = encodeDecreaseLiquidity(decreaseParams)
+
+        const collectParams = {
+          tokenId: tokenId,
+          recipient: bob.address,
+          amount0Max: MAX_UINT128,
+          amount1Max: MAX_UINT128,
+        }
+
+        const encodedCollectCall = encodeCollect(collectParams)
+
+        planner.addCommand(CommandType.V3_POSM_CALL, [encodedDecreaseCall])
+        planner.addCommand(CommandType.V3_POSM_CALL, [encodedCollectCall])
+
+        await executeRouter(planner)
+
+        position = await v3NFTPositionManager.positions(tokenId)
+        let owed0 = position.tokensOwed0
+        let owed1 = position.tokensOwed1
+
+        expect(owed0).to.eq(0)
+        expect(owed1).to.eq(0)
+
+        let bobToken0BalanceAfter = await usdcContract.balanceOf(bob.address)
+        let bobToken1BalanceAfter = await wethContract.balanceOf(bob.address)
+
+        // bob is the recipient - he should have received the owed tokens
+        expect(bobToken0BalanceAfter).to.be.gt(bobToken0BalanceBefore)
```

Comment:
you know exactly how much it should have increased by by looking up `tokensOwed` in the v3 position manager before calling collect

Reply:
The thing is that permit, decrease, and collect are all happening in one call, so i cant look up tokensOwed after decrease and before collect. ill add another test that breaks them up - that way i can check to make sure they are equal

Reply:
hm for some reason tokensOwed1 isnt equaling out